### PR TITLE
Fix lint errors

### PR DESCRIPTION
**Pull request by romanwozniak**
# Context:
In the past, CI pipeline ignored lint errors, which resulted in a large number of such issues accumulated in the codebase of this repo:
Screenshot 2022-10-31 at 12 02 41 PM

This PR fixes all existing lint errors and updates the CI pipeline to use [golangci/golangci-lint ](https://github.com/golangci/golangci-lint) instead of deprecated [golang/lint](https://github.com/golang/lint). Moving forward, any added lint error will cause CI pipeline failure.

### CODE REVIEW

File: api/.gitignore [hunk 1]
```diff
@@ -1,210 +0,0 @@
...
-*.cab
-*.msi
-*.msix
-*.msm
-*.msp
-
-# Windows shortcuts
-*.lnk
-
-# End of https://www.toptal.com/developers/gitignore/api/go,visualstudiocode,intellij,macos,linux,windows
```
Comment (karzuo) [inline api/.gitignore:210]: why is this file removed?

Comment (romanwozniak) [reply api/.gitignore:210]: 1. There is already `.gitignore` at the project root. 
2. This gitignore contains a lot of irrelevant exclusions, that instead should be added to [developer's global gitignore configuration](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration)

**Review by karzuo — APPROVED** [review]
overall LGTM.
Question: 
For camelcase on acronyms like URL, ID, JSON etc, It that the convention we are adopting in DSP or just your personal preference?

**Comment by romanwozniak** [comment]
> overall LGTM. Question: For camelcase on acronyms like URL, ID, JSON etc, It that the convention we are adopting in DSP or just your personal preference?

It's the canonical Golang naming convention
